src/Interpreters/Cache/QueryCache.cpp

```diff
@@ -115,6 +115,15 @@ ASTPtr removeQueryCacheSettings(ASTPtr ast)
     return transformed_ast;
 }
 
+String queryStringFromAst(ASTPtr ast)
+{
+    WriteBufferFromOwnString buf;
+    IAST::FormatSettings format_settings(buf, /*one_line*/ true);
```

Comment:
why not reuse `serializeAST` from `Parsers/formatAST.h`?

Reply:
Interesting, didn't know about this. I'll try to use it.